Disambiguate index property read/write methods correctly (#380)
integrated-application-development/sonar-delphi: CHANGELOG.md

- Name resolution failures when accessing ancestors of enclosing types from nested type methods.
- Name resolution failures on invocations of methods with generic open array parameters.
- Name resolution failures around `Create` calls on types with `constructor` constraints.

> - Name resolution failures on `read` and `write` specifiers for properties with an `index` specifier.

Reviewer: nit;
```suggestion
- Name resolution failures on `read` and `write` specifiers of indexed properties.
```

Author: Fixed
